src/s3.ts

```diff
@@ -312,9 +327,9 @@ export const deleteS3Objects = async (
 
     if (Contents) {
       await Promise.all(
-        Contents.map(c => {
+        Contents.map(async c => {
           // delete each file with given path
-          Private.deleteFile(s3Client, bucketName, c.Key!);
+          await Private.deleteFile(s3Client, bucketName, c.Key!);
```

Comment:
I suggest you `return` this promise isn't of `await` because then `await Promise.all` will await it anyway.